File: Programming with C#/1. C# Fundamentals I/5. Conditional Statements/06. The Biggest of Five Numbers/TheBiggestOfFiveNumbers.cs
/*Problem 6. The Biggest of Five Numbers
Write a program that finds the biggest of five numbers by using only five if statements.*/

using System;

class TheBiggestOfFiveNumbers
{
    static void Main()
    {
        Console.Title = "The Biggest of Five Numbers";

        Console.WriteLine("Enter five number:");
        Console.Write("a --> ");
        double a = double.Parse(Console.ReadLine());
        Console.Write("b --> ");
        double b = double.Parse(Console.ReadLine());
        Console.Write("c --> ");
        double c = double.Parse(Console.ReadLine());
        Console.Write("d --> ");
        double d = double.Parse(Console.ReadLine());
        Console.Write("e --> ");
        double e = double.Parse(Console.ReadLine());
        double biggest = 0;

        if (a > b)
        {
            biggest = a;

            if (c > biggest)
            {
                biggest = c;

            }

            if (d > biggest)
            {
                biggest = d;
            }

            if (e > biggest)
            {
                biggest = e;
            }
        }
        else
        {
            biggest = b;

            if (c > biggest)
            {
                biggest = c;
            }

            if (d > biggest)
            {
                biggest = d;
            }

            if (e > biggest)
            {
                biggest = e;
            }
        }

        Console.WriteLine("Result --> {0}", biggest);
    }
}

File: Web Design &UI/3. JS-Fundamentals/HW/03. Conditional statements/Conditional statements/theBiggestOfFiveNumbers.js
/**
 Problem 7. The biggest of five numbers

 Write a script that finds the greatest of given 5 variables.
 Use nested if statements.
 Examples:

 a    b    c    d    e    biggest
 5    2    2    4    1    5
 -2    -22    1    0    0    1
 -2    4    3    2    0    4
 0    -2.5    0    5    5    5
 -3    -0.5    -1.1    -2    -0.1    -0.1
 */

var arrayA = [5, -2, -2, 0, -3],
    arrayB = [2, -22, 4, -2.5, -0.5],
    arrayC = [2, 1, 3, 0, -1.1],
    arrayD = [4, 0, 2, 5, -2],
    arrayE = [1, 0, 0, 5, -0.1];

for (var i = 0; i < arrayA.length; i++) {
    isBiggest(arrayA[i], arrayB[i], arrayC[i], arrayD[i], arrayE[i]);
}

function isBiggest(a, b, c, d, e) {
    if (a > b) {
        if (a > c) {
            if (a > d) {
                if (a > e) {
                    console.log('Biggest is: ' + a);
                } else {
                    console.log('Biggest is: ' + e);
                }
            } else {
                if (d > e) {
                    console.log('Biggest is: ' + d);
                } else {
                    console.log('Biggest is: ' + e);
                }
            }
        } else {
            if (c > d) {
                if (c > e) {
                    console.log('Biggest is: ' + c);
                } else {
                    console.log('Biggest is: ' + e);
                }
            } else {
                if (d > e) {
                    console.log('Biggest is: ' + d);
                } else {
                    console.log('Biggest is: ' + e);
                }
            }
        }
    } else {
        if (b > c) {
            if (b > d) {
                if (b > e) {
                    console.log('Biggest is: ' + b);
                } else {
                    console.log('Biggest is: ' + e);
                }
            } else {
                if (d > e) {
                    console.log('Biggest is: ' + d);
                } else {
                    console.log('Biggest is: ' + e);
                }
            }
        } else {
            if (c > d) {
                if (c > e) {
                    console.log('Biggest is: ' + c);
                } else {
                    console.log('Biggest is: ' + e);
                }
            } else {
                if (d > e) {
                    console.log('Biggest is: ' + d);
                } else {
                    console.log('Biggest is: ' + e);
                }
            }
        }
    }
}
